keps/sig-auth/740-service-account-external-signing/README.md

```diff
@@ -1,113 +1,560 @@
+   - Compare against `serviceaccount_valid_tokens_total`
+
+
+###### Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?
+
+Will be covered by integration tests.
+
+###### Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?
+
+No.
+
+### Monitoring Requirements
+
+###### How can an operator determine if the feature is in use by workloads?
+
+The Feature would not be used by workload directly but will be used by kube-apiserver.
+
+The usage should be visible to the operator using Audit logs.
+<!-- TODO 
+  Add details on increasing audit log surface area for External signers
+-->
+
+###### How can someone using this feature know that it is working for their instance?
+
+The feature is not used by individual pods but the kube-apiserver itself. Simply successful Token requests and successful auth with kube-apiserver when using a bearer token in the request shall be indicators enough. 
+
+###### What are the reasonable SLOs (Service Level Objectives) for the enhancement?
+
+<!-- TODO
+Needs Benchmarking on SLIs.
+-->
+
+###### What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?
+
+- [x] Metrics
+  - Metric name: `apiserver_request_total` and `apiserver_request_duration_seconds`
+  - Aggregation method:  aggregate over `job="kubernetes-apiservers",group="", version="v1",resource="serviceaccounts",subresource="token"`
+  - Components exposing the metric: kube-apiserver
+
+###### Are there any missing metrics that would be useful to have to improve observability of this feature?
+
+1. serviceaccount_jwks_freshness
+2. serviceaccount_oidc_jwks_freshness
+3. serviceaccount_jwks_fetch_requests_total
+
+- 1 and 2 can both be gauge metrics. 
+- Each will indicate how long it's been since the supported key set was last synced.
+- 3 can be a cumulatively increasing metric with `status` that will also enable deriving error rate.
+
+### Dependencies
+
+One new dependency will be introduced and it will only be required for clusters configured/opted-in via the `--service-account-signing-endpoint` flag.
+
+###### Does this feature depend on any specific services running in the cluster?
+
+The feature depends on a cluster level service that will implement [ExternalJWTSigner RPC](#externaljwtsigner-rpc) and will serve on UDS. each distribution will have their own version, so let's just address it as `ExternalJWTSigner`.
+
+- ExternalJWTSigner
+    - Usage description:
+      This service will act as a client to the external key management solution that a distribution will integrate with.
+
+      NOTE: this being a configurable/opt-in feature, the impact will only be seen on clusters that are using the feature.
+
+      - Impact of its outage on the feature:
+        - kube-apiserver will not be able to sign new service account JWTs.
+        - old JWTs will continue to be supported since we have an in cluster cache.
```

Comment:
What will API server do if the cache expiration/suggested refresh time passes and can't be updated? Continue to use the old signing keys indefinitely?

Reply:
I want to say that It will continue operating with stale set of Keys. But now that I think about it, shall it have a TTL so we don't keep supporting a set of keys indefinitely? if so, how do we determine the TTL?

Reply:
> What will API server do if the cache expiration/suggested refresh time passes and can't be updated? Continue to use the old signing keys indefinitely?

For verifying, yes.

new tokens signed by new `kid`s would fail if new public keys can't be fetched

Reply:
Corrected with mentions of 'existing keys will continue to be supported'.